packages/fxa-settings/src/pages/Signup/container.tsx

```diff
@@ -75,22 +99,76 @@ const SignupContainer = ({
             hardNavigateToContentServer(
               `/signin?email=${queryParamModel.email}`
             );
+            // TODO: Probably move this to the Index page onsubmit once
+            // the index page is converted to React, we need to run it in
+            // signup and signin for Sync
+          } else if (isSyncWebChannel) {
+            firefox.fxaCanLinkAccount({ email: queryParamModel.email });
           }
+        } else if (isSyncWebChannel) {
+          // TODO: Probably move this to the Index page onsubmit once
+          // the index page is converted to React, we need to run it in
+          // signup and signin for Sync
+          firefox.fxaCanLinkAccount({ email: queryParamModel.email });
         }
       }
       setShowLoadingSpinner(false);
     })();
   });
 
+  useEffect(() => {
+    // This sends a webchannel message to the browser to prompt a response
+    // that we listen for.
+    // TODO: In content-server, we send this on app-start for all integration types.
+    // Do we want to move this somewhere else once the index page is Reactified?
+    if (isSyncWebChannel) {
+      firefox.addEventListener(
+        FirefoxCommand.FxAStatus,
+        handleFxAStatusSyncEngineEvent
+      );
+
+      // requestAnimationFrame ensures the event listener is added first
+      // otherwise, there is a race condition
+      requestAnimationFrame(() =>
+        firefox.send(FirefoxCommand.FxAStatus, {
+          // TODO: Improve getting 'context', probably set this on the integration
+          context: isSyncDesktop
+            ? Constants.FX_DESKTOP_V3_CONTEXT
+            : Constants.OAUTH_CONTEXT,
+          isPairing: false,
+          service: Constants.SYNC_SERVICE,
+        })
+      );
+    }
+  });
+
+  const handleFxAStatusSyncEngineEvent = (event: any) => {
```

Comment:
What are your thoughts on all the event handling being inside the CWTS component itself? Perhaps passing in the integration and having that component listen and setup everything it needs?

Reply:
@vbudhram That's probably a good idea. 👍 My thought process here was that I originally had these inside Signup instead of its container, but that didn't feel like the right place and I needed to do some `syncMobileWebchannel` checks anyway in the container. Since the fired events/listeners were in `app-start.js` in content-server I also thought we might end up moving them higher up somewhere since I know FxAStatus is commonly listened for and responded to, but I like the idea of (at least most things) CWTS related being inside CWTS.

I think we can file an issue to account for the FF Sync desktop changes, and look at this then.

Reply:
Mentioned this in: https://mozilla-hub.atlassian.net/browse/FXA-8671